LogicLayer/GameEventInvitations/GameEventInvitationsService.cs

```diff
@@ -0,0 +1,106 @@
+﻿using GameBoard.DataLayer.Entities;
+using GameBoard.DataLayer.Enums;
+using GameBoard.DataLayer.Repositories;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GameBoard.LogicLayer.GameEventInvitations.Dtos;
+using GameBoard.LogicLayer.Notifications;
+
+namespace GameBoard.LogicLayer.GameEventInvitations
+{
+    internal class GameEventInvitationsService : IGameEventInvitationsService
+    {
+        private readonly IGameBoardRepository _repository;
+        private readonly IMailSender _mailSender;
+
+        public GameEventInvitationsService(IGameBoardRepository repository, IMailSender mailSender)
+        {
+            _repository = repository;
+            _mailSender = mailSender;
+        }
+
+        private IQueryable<GameEventParticipation> GetGameEventParticipationsInOneEvent(
+            int gameEventId,
+            [NotNull] string userName)
+        {
+            return _repository.ApplicationUsers
+                .Where(ApplicationUser.UserNameEquals(userName))
+                .Include(u => u.Participations)
+                .SelectMany(u => u.Participations)
+                .Where(p => p.TakesPartInId == gameEventId);
+        }
+
+        private async Task ChangeGameEventInvitationStatusAsync(
+            int gameEventId,
+            [NotNull] string invitedUserName,
+            ParticipationStatus participationStatus)
+        {
+            var participation = await
+                GetGameEventParticipationsInOneEvent(gameEventId, invitedUserName)
+                .SingleAsync(p => p.ParticipationStatus == ParticipationStatus.PendingGuest);
+
+            participation.ParticipationStatus = participationStatus;
+
+            await _repository.SaveChangesAsync();
+        }
+
+        public Task RejectGameEventInvitationAsync(int gameEventId, string invitedUserName)
+        {
+            return ChangeGameEventInvitationStatusAsync(
+                gameEventId,
+                invitedUserName,
+                ParticipationStatus.RejectedGuest);
+        }
+
+        public Task AcceptGameEventInvitationAsync(int gameEventId, string invitedUserName)
+        {
+            return ChangeGameEventInvitationStatusAsync(
+                gameEventId,
+                invitedUserName,
+                ParticipationStatus.AcceptedGuest);
+        }
+
+        public async Task SendGameEventInvitationAsync(CreateGameEventInvitationDto gameEventInvitationDto)
```

Comment:
Did it.

Reply:
The `switch` statement in this method looks bad. First of, you're throwing the `ArgumentOutOfRangeException` without a message, but that's minor. What's worse is that you're throwing that on `default` without handling `Creator`, so if a user tries to invite himself they'll get a 500 and "Unknown error has occured". They should get "You cannot invite yourself" or something similar.

I know it's actually impossible right now for us to invite ourselves via the app because of how frontend works, but that's irrelevant to the logic of this method.

Reply:
Done.

Reply:
"Does not include this value" doesn't tell us what the value was.

Reply:
Fixed.